### ISSUE TITLE
vertical-module

### ISSUE THREAD

**Issue by iamnizzam** [meanjs/generator-meanjs#119]
how to generate vertical module?
like before yo meanjs:vertical-module
not working now...?

**Comment by jrbenny35** [comment]
I have this same problem, just getting the following

yo meanjs:vertical-module projects
undefined
You're using the official MEAN.JS generator.

**Comment by iamnizzam** [comment]
i want to use MEAN.JS generator for 0.3.3... its awsome

**Comment by codydaig** [comment]
The generator just went through an overhaul to be inline with the current meanjs project which is on version 0.4.2. The sub generators are under development and not included in the latest release. If you want to use 0.3.3 and the vertical sub generator you'll have to install the 0.1.x version of the generator.

**Comment by iamnizzam** [comment]
npm install generator-mean@0.1.x
but still not workin...

**Comment by jrbenny35** [comment]
This bit of code has worked for me

npm install meanjs/generator-meanjs#0.4-dev

Found here: https://groups.google.com/forum/#!topic/meanjs/wKavBQOCZvw

**Comment by mleanos** [comment]
@iamnizzam @jrbenny35 The meanjs yeoman generator has been updated to the most recent meanjs release 0.4.1. https://www.npmjs.com/package/generator-meanjs

**Comment by iamnizzam** [comment]
thanks  @jrbenny35 ...
its work

**Comment by RandyDavis** [comment]
i've gone through all of the steps @jrbenny35 and @iamnizzam listed and I still get "undefined" after trying to install a vertical module.  no matches found: meanjs/generator-meanjs#0.4-dev is returned after i try running npm install meanjs/generator-meanjs#0.4-dev as well.  any other ideas?

**Comment by ilanbiala** [comment]
Don't install 0.4-dev, you should just install v0.4.1

**Comment by RandyDavis** [comment]
I went back and redid my app using v0.4.1 from scratch as I originally started with v0.4.0 and got to the same point and whenever I run "yo meanjs:vertical-module <module-name>" it behaves like it's trying to create a new meanjs app and responds "undefined" first and then with "What mean.js version would you like to generate?..."  Where am I going wrong here?

**Comment by ilanbiala** [comment]
@RandyDavis vertical modules don't exist right now. We currently only have the generator for the main app set up for 0.4.x.

**Comment by RandyDavis** [comment]
So what is the exact command that I need to use to create a new module? Just want to make sure I don't mess this up as I assume that I don't use "crud-module" anymore. Thanks!

On Sep 21, 2015 10:19 AM, Ilan Biala <notifications@github.com> wrote:

@RandyDavis<https://github.com/RandyDavis> vertical modules don't exist right now. We currently only have the generator for the main app set up for 0.4.x.

**Comment by ilanbiala** [comment]
@RandyDavis that feature doesn't exist yet for 0.4.x. Feel free to set up a PR for it.

**Comment by RandyDavis** [comment]
Oh so if I want to use meanjs and create modules, I need to rollback to 0.3.x? If so, then that makes sense now.

On Sep 21, 2015 10:59 AM, Ilan Biala <notifications@github.com> wrote:

@RandyDavis<https://github.com/RandyDavis> that feature doesn't exist yet for 0.4.x. Feel free to set up a PR for it.

**Comment by ilanbiala** [comment]
Yes, though hopefully we'll have module generation out very soon, so maybe start working on the basics of it and I'll try and finish it up this weekend.

**Comment by RandyDavis** [comment]
Thanks for the clarification! Looking forward to trying it out when it's ready.

On Sep 21, 2015 11:23 AM, Ilan Biala <notifications@github.com> wrote:

Yes, though hopefully we'll have module generation out very soon, so maybe start working on the basics of it and I'll try and finish it up this weekend.

**Comment by RagingAlcoholic** [comment]
Hey guys. Found this quick fix on SO that worked for me. Let me know if it helps!

**Comment by iamnizzam** [comment]
is module generation finish yet @ilanbiala ?

**Comment by ilanbiala** [comment]
@iamnizzam no.

**Comment by MichaelGatesDev** [comment]
Will rolling my project back from 0.4.1 to 0.3.x destroy my project? I'm afraid all the hours I've put in are now gone if that's so.

**Comment by ilanbiala** [comment]
@MichaelGatesDev It depends on what you have done with it. You should try to use 0.4.1, it has lots of goodies. :)

**Comment by MichaelGatesDev** [comment]
I would love to except it doesn't have vertical module support yet? That's
exactly what I'm needing atm. I wanna generate a new CRUD module

**Comment by englishandy** [comment]
any updates on the fix? i unable to do much without the CRUD module generator.

**Comment by pateldeepp** [comment]
Hello,
This is back burning issue for many users. Please fix this ASAP cause I have seen people do migrate to Mean.IO after trying a lot to work on CRUD Module. Issue seems there for 3 of my team-mates we in different environment.

**Comment by MichaelGatesDev** [comment]
I hope this is one of their top priorities. I'll also be migrating to MEAN.IO because I've gotten this far and now I have to wait (time is money) or downgrade. It's my fault for not reading beforehand, but still.

**Comment by codydaig** [comment]
We have an open PR that we are in the code review process for. I hope to see it merged in within a couple days. That PR works as is so you can use that Branch to generate vertical-modules until it's merged in.

**Comment by emathias** [comment]
Any news on that ?  Maybe an alternative solution in the meantime ?

**Comment by aquadestructor** [comment]
After hours of haunting errors I managed to find a way to generate crud module. As you guys know 0.4.x generators are still under development so there's a way to create another meanjs project folder with 0.3 version. You can downgrade to meanjs 0.3x without destroying the meanjs 0.4 project folder. All you have to do is to downgrade generator-meanjs to 0.1.12 version and downgrade yo to 1.3.3 or the versions till january 2015. Let me know if you get stuck into some issues.

**Comment by iamnizzam** [comment]
any update @ilanbiala

**Comment by ilanbiala** [comment]
@iamnizzam #131

**Comment by hafsteinn** [comment]
Any news on the vertical module generator ? 

Using 0.4.1 and doing 'yo meanjs:vertical-module asdf' results in yo trying to scaffold a brand new project ...

**Comment by ilanbiala** [comment]
@hafsteinn #131 is still open and we need to finish that up.

**Comment by matenoz** [comment]
A rude bash script (workaroud for linux user):
1) create a template directory (see below) in the app's root:
 template/
├── client
│   ├── client.module.js
│   ├── config
│   │   ├── client.config.js
│   │   └── client.routes.js
│   ├── controllers
│   │   └── client.controller.js
│   ├── services
│   │   └── client.service.js
│   └── views
│       ├── create.client.view.html
│       ├── edit.client.view.html
│       ├── list.client.view.html
│       └── view.client.view.html
├── server
│   ├── config
│   │   └── server.config.js
│   ├── controllers
│   │   └── server.controller.js
│   ├── models
│   │   └── server.model.js
│   ├── policies
│   │   └── server.policy.js
│   └── routes
│       └── server.routes.js
└── tests
    ├── client
    │   └── client.controller.tests.js
    ├── e2e
    │   └── e2e.tests.js
    └── server
        ├── server.model.tests.js
        └── server.routes.tests.js
template/
├── client
│   ├── client.module.js
│   ├── config
│   │   ├── client.config.js
│   │   └── client.routes.js
│   ├── controllers
│   │   └── client.controller.js
│   ├── services
│   │   └── client.service.js
│   └── views
│       ├── create.client.view.html
│       ├── edit.client.view.html
│       ├── list.client.view.html
│       └── view.client.view.html
├── server
│   ├── config
│   │   └── server.config.js
│   ├── controllers
│   │   └── server.controller.js
│   ├── models
│   │   └── server.model.js
│   ├── policies
│   │   └── server.policy.js
│   └── routes
│       └── server.routes.js
└── tests
    ├── client
    │   └── client.controller.tests.js
    ├── e2e
    │   └── e2e.tests.js
    └── server
        ├── server.model.tests.js
        └── server.routes.tests.js

then 
in the root folder run
$ mv rude-v-mod-generator.txt rude-v-mod-generator.sh
$ sh rude-v-mod-generator.sh  my_module_name
(Test it before run)

[rude-v-mod-generator.txt](https://github.com/meanjs/generator-meanjs/files/44517/rude-v-mod-generator.txt)

**Comment by tbaustin** [comment]
I am getting this error when I run yo meanjs:vertical-module customers : 
Error meanjs:vertical-module customers

You don't seem to have a generator with the name meanjs:vertical-module installed.
You can see available generators with npm search yeoman-generator and then install them with npm install [name].
To see the 2 registered generators run yo with the `--help` option.

**Comment by jpimentel** [comment]
Im getting the same error as @tbaustin

**Comment by ilanbiala** [comment]
This feature hasn't been added yet, we'll post on this thread once we have released new subgenerators.

**Comment by Siddharth-ss42279** [comment]
same error as of @tbaustin  , any solution ??

**Comment by ilanbiala** [comment]
Try `mean-module`, which is the first half of the vertical module generator when you install 0.4.3. We are working on getting the second half in as part of 0.4.4.

**Comment by amercode** [comment]
I came up with a hack to fix this issue: please check out my video:
https://www.youtube.com/watch?v=Nkpdp1ofux8

**Comment by eliavmaman** [comment]
also need it please.. :)

**Comment by amercode** [comment]
This is my solution: https://www.youtube.com/watch?v=Nkpdp1ofux8
